core/domain/topic_fetchers.py

```diff
@@ -747,3 +748,30 @@ def get_canonical_story_dicts(
         canonical_story_dicts.append(story_summary_dict)
 
     return canonical_story_dicts
+
+
+def get_all_story_exploration_ids(
+    topic_name: Optional[str] = None
```

Comment:
Use topic ID. Always use IDs where possible, especially for args that are passed around.

Reply:
Done. Note that these changes were moved to `topic_services.py`.